File: src/vovka_pkg/scripts/find_max_client.py
#!/usr/bin/env python

import sys
import rospy
from vovka_pkg.srv import *


def find_max_client(values):
    rospy.wait_for_service('find_max')
    try:
        find_max = rospy.ServiceProxy('find_max', FindMax)
        response = find_max(values)
        return response.max_value
    except rospy.ServiceException as e:
        print 'Service call failed: %s' % e


def usage():
    return '%s x1 x2 x3 ...' % sys.argv[0]


if __name__ == '__main__':
    if len(sys.argv) <= 2:
        rospy.loginfo(usage())
        sys.exit(1)
    else:
        values = map(int, sys.argv[1:])
    print 'Request find max among [%s]' % str(values)
    print 'Max is %d' % find_max_client(values)

File: src/vovka_pkg/src/find_max_client.cpp
#include "ros/ros.h"
#include "vovka_pkg/FindMax.h"

#include <cstdlib>

int main(int argc, char **argv) {

    ros::init(argc, argv, "find_max_client");
    if (argc <= 2) {
        ROS_INFO("usage: find_max_client x1 x2 x3 x4 ...");
        return 1;
    }

    ros::NodeHandle node;
    ros::ServiceClient client = node.serviceClient<vovka_pkg::FindMax>("find_max");

    vovka_pkg::FindMax srv;
    for (int i = 1; i < argc; ++i) {
        srv.request.values.push_back(atoll(argv[i]));
    }

    if (client.call(srv)) {
        ROS_INFO("Max is %d", (int)srv.response.max_value);
    } else {
        ROS_ERROR("Failed to call service find_max");
        return 1;
    }

    return 0;
}
